jwql/instrument_monitors/common_monitors/readnoise_monitor.py

```diff
@@ -512,24 +516,24 @@ def process(self, file_list):
                                   'readnoise_filename': os.path.basename(readnoise_outfile),
                                   'full_image_mean': float(full_image_mean),
                                   'full_image_stddev': float(full_image_stddev),
-                                  'full_image_n': full_image_n.astype(float),
-                                  'full_image_bin_centers': full_image_bin_centers.astype(float),
+                                  'full_image_n': list(full_image_n.astype(float)),
```

Comment:
Do you mind reminding me why we need to type cast these objects into lists? Was this something that sqlalchemy did for us that the django models do not?

Reply:
Exactly. Before I did this, django was throwing `django.db.utils.ProgrammingError: can't adapt type 'numpy.ndarray'`, which seems like a direct translation of an error from postgresql, `psycopg2.ProgrammingError: can't adapt type 'numpy.ndarray'`. My guess is that SQLAlchemy had a basic adaptation where it would automatically turn something into a list if it wasn't a known type but it had a built-in way to do that (I'm pretty sure that in python you can have an object property that defines whether it's list-like, so my guess is that django just doesn't translate object types as they go to the database.

Reply:
Cool, I figured there was a good reason thank you!